### ISSUE TITLE
mpatibility with SD.Next from VLAD

### ISSUE THREAD

**Issue by Centurion-Rome** [namkazt/sd-webui-prompt-history#16]
This is great extension and working very well with A1111, sadly I am getting error when using it with SD.Next Vlads fork from A1111.

```
20:40:48-409139 ERROR    gradio call: AttributeError
╭────────────────────────────────────────────────────────── Traceback (most recent call last) ───────────────────────────────────────────────────────────╮
│ F:\!AI\StableDiffusion\automatic\modules\call_queue.py:34 in f                                                                                         │
│                                                                                                                                                        │
│    33 │   │   │   try:                                                                                                                                 │
│ ❱  34 │   │   │   │   res = func(*args, **kwargs)                                                                                                      │
│    35 │   │   │   │   progress.record_results(id_task, res)                                                                                            │
│                                                                                                                                                        │
│ F:\!AI\StableDiffusion\automatic\modules\txt2img.py:56 in txt2img                                                                                      │
│                                                                                                                                                        │
│   55 │   if processed is None:                                                                                                                         │
│ ❱ 56 │   │   processed = processing.process_images(p)                                                                                                  │
│   57 │   p.close()                                                                                                                                     │
│                                                                                                                                                        │
│ F:\!AI\StableDiffusion\automatic\extensions\sd-webui-prompt-history\lib_history\image_process_hijacker.py:25 in process_images                         │
│                                                                                                                                                        │
│   24 │   # add result to history                                                                                                                       │
│ ❱ 25 │   global_state.add_config( uuid.uuid4().hex, res.prompt[:64], shared.opts.sd_model_che                                                          │
│   26 │   return res                                                                                                                                    │
│                                                                                                                                                        │
│ F:\!AI\StableDiffusion\automatic\extensions\sd-webui-prompt-history\scripts\prompt_history_script.py:76 in add_config                                  │
│                                                                                                                                                        │
│    75 │   │   img = img.resize((new_width, new_height), Image.LANCZOS)                                                                                 │
│ ❱  76 │   │   images.save_image_with_geninfo(img, None, os.path.join(global_state.history_path                                                         │
│    77 │   else:                                                                                                                                        │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: module 'modules.images' has no attribute 'save_image_with_geninfo'
```

please add support for vlads SD.Next: https://github.com/vladmandic/automatic

**Comment by GraspGG** [comment]
__As of here: [vladmantic automatic_SD.NEW issue and comment about this](https://github.com/vladmandic/automatic/issues/1596)__

> you have not posted actual trace of the error, only noted it in a headline.
> 
> **function `modules.images.save_image_with_geninfo` was obsoleted a long time ago and its not referenced anywhere.**
> 
> which means you have an **_extension_** that is trying to use it. well, that function was always internal-only and no extension should use it. if **_extension_** wants to use a correct method, it should rely on `modules.images.save_image`.
> 
> i suggest to find which **_extension_** is doing this (you'll see it in console log showing actual error) and report to extension owner.

Where all **_extension_** references are pointing to **sd-webui-prompt-history**
...if that is of any help (:

**Comment by namkazt** [comment]
@GraspGG  thanks for the information

**Comment by namkazt** [comment]
the fix is in this commit: https://github.com/namkazt/sd-webui-prompt-history/commit/b980c5c522abe891aea911aa5c6bab963f5a0c91

please update and test again as i;m not using vlad version.

**Comment by namkazt** [comment]
feel free to open an issue if it still persist

**Comment by GraspGG** [comment]
> @GraspGG thanks for the information

@namkazt you're welcome
Thanks for the fix!! (:

> the fix is in this commit: [b980c5c](https://github.com/namkazt/sd-webui-prompt-history/commit/b980c5c522abe891aea911aa5c6bab963f5a0c91)
> 
> please update and test again as i;m not using vlad version.

Updated - Enabled - Tested:
- No more errors after image generation - check
- Prompt has been saved and is shown/listed on the "Prompt History" tab afterwards- check
- On SD.NEXT/A1111-"Prompt History" tab there are tiny little graphic issues - but well, i couldn't care less - that whole project frontend is kinda new/different. So i guess noone can even say what is the reason for those little misalignments (: 

If you would like a screenshot showing the "Prompt History" tab, let me know - i would have placed it right here, hidden below a "spoiler" but couldn't find one. Placing a screenshot here, big and omnipresent, would make that topic multiple times bigger than it is ^^)

Thanks again,
for the extension itself and for the quick fix!
Take care
